DiceKit/ProbabilityMass.swift

```diff
@@ -0,0 +1,104 @@
+//
+//  ProbabilityMass.swift
+//  DiceKit
+//
+//  Created by Brentley Jones on 7/19/15.
+//  Copyright © 2015 Brentley Jones. All rights reserved.
+//
+
+import Foundation
+
+public struct ProbabilityMass: ApproximatelyEquatable {
+    
+    public typealias Value = FrequencyDistribution.Value
+    public typealias Probability = Double
+    
+    public static let defaultProbabilityEqualityDelta = 1e-16 // This is for 64-bit. What should 32-bit be?
+    public static var probabilityEqualityDelta = defaultProbabilityEqualityDelta
+    
+    public static let additiveIdentity = ProbabilityMass(0)
+    
+    public let frequencyDistribution: FrequencyDistribution
+    
+    public func normalize() -> ProbabilityMass {
+        return ProbabilityMass(frequencyDistribution, normalize: true)
+    }
+    
+    internal init(_ frequencyDistribution: FrequencyDistribution, normalize: Bool) {
+        self.frequencyDistribution = normalize ? frequencyDistribution.normalizeOutcomes() : frequencyDistribution
+    }
+    
+    public init(_ frequencyDistribution: FrequencyDistribution) {
+        self.init(frequencyDistribution, normalize: true)
+    }
+    
+    public init(_ constant: Int) {
+        self.init(FrequencyDistribution([constant : 1]), normalize: false)
+    }
+    
+    public subscript(value: Value) -> Probability? {
+        get {
+            return frequencyDistribution[value]
+        }
+    }
+    
+}
+
+// MARK: - Equatable
+
+public func == (lhs: ProbabilityMass, rhs: ProbabilityMass) -> Bool {
+    return lhs.frequencyDistribution == rhs.frequencyDistribution
+}
+
+// MARK: - ApproximatelyEquatable
+
+public func ≈ (lhs: ProbabilityMass, rhs: ProbabilityMass) -> Bool {
+    return lhs.frequencyDistribution.approximatelyEqual(rhs.frequencyDistribution, delta: ProbabilityMass.probabilityEqualityDelta)
+}
+
+// MARK: - Operators
+
+extension ProbabilityMass {
+    
+    public func negate() -> ProbabilityMass {
+        let freqDist = frequencyDistribution.negateValues()
+        
+        return ProbabilityMass(freqDist, normalize: false)
+    }
+    
+    public func or(x: ProbabilityMass) -> ProbabilityMass {
+        let freqDist = frequencyDistribution.add(x.frequencyDistribution)
+        
+        return ProbabilityMass(freqDist, normalize: true)
+    }
+    
+    public func and(x: ProbabilityMass) -> ProbabilityMass {
+        let freqDist = frequencyDistribution.multiply(x.frequencyDistribution)
+        
+        return ProbabilityMass(freqDist, normalize: true)
+    }
+    
+    // Not sure on this name
+    public func product(x: ProbabilityMass) -> ProbabilityMass {
```

Comment:
That's tricky, because all of them are hard to explain. But maybe:

> Given the chance of a probability mass, for each of it's outcomes (the sums in die rolls) `and` the other probability mass against itself, scaling by the chance of the outcome, and finally `and` all of those respective results together. Essentially, the probability mass of something like `(1d2)d6` happening.